first_hour_on_vwb/working_with_resources.ipynb

```diff
@@ -0,0 +1,1436 @@
+    "\n",
+    "\n",
+    "# Instantiate widget\n",
+    "add_repo_widget = AddRepoWidget()\n",
+    "display(add_repo_widget.vb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 2. Add a BigQuery dataset\n",
+    "\n",
+    "Run the cell below to create a widget that adds a BigQuery dataset as a referenced resource to your workspace.\n",
+    "\n",
+    "Then, populate the input fields:\n",
+    "\n",
+    "- `Cloning`: Read about these options [in the Cloning Instruction section](#choosing-the-right-cloning-instruction).\n",
+    "- `Project ID`: The Google Cloud Project ID of your dataset.\n",
+    "- `Dataset ID`: The BigQuery dataset ID of your dataset.\n",
+    "- `Path`: Consists of the project ID and dataset ID with format `<PROJECT_ID>.<DATASET_ID>`.\n",
+    "- `Name`: The name by which you will reference this resource in your workspace. A short and descriptive name is suggested, e.g., `human-genomic-data`.\n",
+    "- `Description`: A description of this resource's contents and purpose.\n",
+    "\n",
+    "**NOTE:** You must populate either\n",
+    "\n",
+    "- the path field (which will have format `<PROJECT_ID>.<DATASET_ID>`) field or\n",
+    "- both [dataset ID and the project ID](https://cloud.google.com/bigquery/docs/datasets-intro#datasets) fields.\n",
+    "\n",
+    "Click the button to add the BigQuery dataset reference to your workspace. The output should resemble:\n",
+    "\n",
+    "```\n",
+    "Successfully added referenced BigQuery dataset.\n",
+    "Name:         <NAME>\n",
+    "Description:  <DESCRIPTION>\n",
+    "Type:         BQ_DATASET\n",
+    "Stewardship:  REFERENCED\n",
+    "Cloning:      COPY_REFERENCE\n",
+    "Properties:   class Properties {\n",
+    "    []\n",
+    "}\n",
+    "GCP project id: <PROJECT_ID>\n",
+    "BigQuery dataset id: <DATASET_ID>\n",
+    "Location: US\n",
+    "# Tables: <NUMBER_OF_TABLES>\n",
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "class AddBQDatasetWidget(object):\n",
+    "    def __init__(self):\n",
+    "        self.label = widgets.Label(\n",
+    "            value=\"Please provide appropriate values in the input boxes.\")\n",
+    "        self.cloning_drop_down = DropdownInputWidget(\n",
+    "            ['COPY_NOTHING', 'COPY_DEFINITION', 'COPY_RESOURCE', 'COPY_REFERENCE'], 'COPY_REFERENCE', \"Cloning:\").get()\n",
+    "        self.input_dataset_id = TextInputWidget(\n",
+    "            \"<DATASET_ID>\", \"Dataset ID:\").get()\n",
+    "        self.input_description = TextInputWidget(\n",
+    "            \"<REPO_DESCRIPTION>\", \"Description:\").get()\n",
+    "        self.input_name = TextInputWidget(\"<NAME>\", \"Name:\").get()\n",
+    "        self.input_path = TextInputWidget(\"<PATH>\", \"Path:\").get()\n",
```

Comment:
Can the UI make it clear which that either path field OR (dataset id AND project ID) must be populated? As is it feels like I need to fill out all fields. Even minimally just some explanation text under `Please provide appropriate values in the input boxes.` would suffice.

Reply:
addressed!